### ISSUE TITLE
[BUG] evaluator_config={"average":None} not working in mlflow.evaluate() for multiclass classification 

### ISSUE THREAD

**Issue by RRRen94** [mlflow/mlflow#10902]
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Local machine

### Willingness to contribute

No. I cannot contribute a bug fix at this time.

### MLflow version

2.9.2

### System information

databricks

### Describe the problem

For multiclass classification problem, if I use `evaluator_config = {"average": None}` to set the averaging method to use when computing classification metrics, I will get a list of float `[0.7, 0.8, 0.6]` as my metrics for each class. But this metric list can not pass validation check, as it expect only numerical values as metrics not a list.
https://github.com/mlflow/mlflow/blob/c43823750bffa5b6abcc086683b15a068513b67b/mlflow/utils/validation.py#L137 

### Code to reproduce issue

```
with mlflow.start_run() as run:
    # Evaluate the static dataset without providing a model
    result = mlflow.evaluate(
        data=eval_data,
        targets="label",
        predictions="predictions",
        model_type="classifier",
        evaluator_config={"average": None}
```

### Stack trace

```
def _validate_metric(key, value, timestamp, step):
    """
    Check that a metric with the specified key, value, timestamp, and step is valid and raise an
    exception if it isn't.
    """
    _validate_metric_name(key)
    # value must be a Number
    # since bool is an instance of Number check for bool additionally
    if not _is_numeric(value):
        raise MlflowException(
            f"Got invalid value {value} for metric '{key}' (timestamp={timestamp}). "
            "Please specify value as a valid double (64-bit floating point)",
            INVALID_PARAMETER_VALUE,
        )
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [X] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [X] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by serena-ruan** [comment]
@RRRen94 Could you paste the full stacktrace where it triggers _validate_metric?

**Comment by RRRen94** [comment]
@serena-ruan Hey, full stacktrace is now updated.

**Comment by serena-ruan** [comment]
@RRRen94 I think the original design is to use `weighted` if "average" method is not set (or None), so to fix the bug I would set "average" to "weighted" if it's None. But for your use case, do you want to log the metric value of each class specifically? Otherwise the workaround is just to remove evaluator_config when calling `.evaluate`

**Comment by RRRen94** [comment]
> @RRRen94 I think the original design is to use `weighted` if "average" method is not set (or None), so to fix the bug I would set "average" to "weighted" if it's None. But for your use case, do you want to log the metric value of each class specifically? Otherwise the workaround is just to remove evaluator_config when calling `.evaluate`

Yes, I want to directly see the metrics value of each class specifically, not weighted or averaged. These single values of each class can be found now in `per_class_metrics.csv` as artifact. But it could be nice to also have these values as metrics.

**Comment by serena-ruan** [comment]
Then I think the best effort would be log each class's metric value separately as now mlflow metric doesn't support lists.

**Comment by serena-ruan** [comment]
cc @prithvikannan WDYT?
